### DISCUSSION TITLE
Changing Http Status Codes

### DISCUSSION THREAD

**Discussion by EAGC06** [FirelyTeam/firely-net-sdk#2075] [Q&A] [answered]
We are always getting a HttpStatusCode of 200 in a Firely Server response even when we throw a FhirOperationException with a different HttpStatusCode, 

```
namespace Hl7.Fhir.Rest
{
    public class FhirOperationException : Exception
    {
        public FhirOperationException(string message, HttpStatusCode status);
    }
}
```

We are implementing it as simple as this:
`throw new FhirOperationException(errorMessage, System.Net.HttpStatusCode.NotFound);`

Is this the correct way to change a HttpStatusCode for a Firely Server response? or how should we be doing it?

**Comment by mbaltus** [comment, marked as answer]
Note that Firely Server is not a part of the Firely .Net SDK, but a separate project that uses the SDK for parts of it.
The answer to how to get the Firely Server to change its response will depend on what you are trying to do:
- If you are creating a Facade on your own database, the outcome should already be 404 when your code does not result in any resources for a read/search request. Firely Server will take care of that. See https://github.com/FirelyTeam/Vonk.Facade.Starter for an example facade implementation.
- If you're creating a plugin for Firely Server, you can set http status codes and outcomes on the response in the VonkContext. See example code here: https://github.com/FirelyTeam/Vonk.Plugin.DocumentOperation/blob/develop/Vonk.Plugin.DocumentOperation/DocumentService.cs.

**Reply by EAGC06** [reply]
Thanks @mbaltus

At the end we were the responsible of those 200 statuses. We removed the conflictive code and start exploring the plugins and, so far, it seems that we found the solution.
